src/file-manager/fm-desktop-icon-view.c

```diff
@@ -387,13 +409,21 @@ realized_callback (GtkWidget *widget, FMDesktopIconView *desktop_icon_view)
 {
     GdkWindow *root_window;
     GdkScreen *screen;
+    GdkDisplay *display;
     GtkAllocation allocation;
-    gint scale;
 
     g_return_if_fail (desktop_icon_view->priv->root_window == NULL);
 
     screen = gtk_widget_get_screen (widget);
-    scale = gtk_widget_get_scale_factor (widget);
+    display = gdk_screen_get_display (screen);
+    if (GDK_IS_X11_DISPLAY (display))
+        root_window = gdk_screen_get_root_window (screen);
+
+    else
+    {
+        /*Do not use the unavailable root window*/
+        desktop_icon_view->priv->root_window = NULL;
+    }
```

Comment:
This is redundant with what you have below L457+, you can just remove it I think (and maybe move the Wayland comment down there)

Reply:
Just saw that and removed the redundent block